enclave-runner/src/usercalls/mod.rs

```diff
@@ -294,6 +301,44 @@ pub(crate) struct EnclaveState {
     last_fd: AtomicUsize,
     exiting: AtomicBool,
     usercall_ext: Box<UsercallExtension>,
+    work_sender: Mutex<Option<crossbeam::crossbeam_channel::Sender<Work>>>,
+}
+
+enum Work {
+    Stopped(StoppedTcs, EnclaveEntry),
+    Running(RunningTcs, tcs::Usercall<ErasedTcs>, (u64,u64), EnclaveEntry),
+}
+
+impl Work {
+    fn do_work(self, io_send_queue: &mpsc::Sender<UsercallSendData>
+    ) -> StdResult<(), mpsc::SendError<UsercallSendData>> {
```

Comment:
I don't think you need to return an error here, the only possible error is that the main event loop has already exited.